### ISSUE TITLE
Script to read srv format.

### ISSUE THREAD

**Issue by kenorb** [FX31337/FX-BT-Scripts#64]
Extend [`read_mt_formats.py`](https://github.com/FX31337/FX-BT-Scripts/blob/master/read_mt_formats.py) to read srv format.

Samples: [srv-files.zip](https://github.com/FX31337/FX-BT-Scripts/files/288393/srv-files.zip)

Suggested command to compare two binary files:

```
diff -y <(xxd FXCM-EURDemo01.srv) <(xxd FXCM-GBPDemo01.srv) | colordiff
```

Est. 4-8h
##

**Comment by ryu0** [comment]
Does this also still need to be resolved?

**Comment by kenorb** [comment]
Yes, please.

**Comment by kenorb** [comment]
If you got issues, the bounty can be increased.

**Comment by ryu0** [comment]
Do you know of some kind of reference material for this SRV file format? It appears to be a "server file" for some kind of software called Metatrader, but I am unable to find any information so far about how this data file is serialized.

**Comment by kenorb** [comment]
It's rather not documented, I'll try to answer that after 1h

**Comment by ryu0** [comment]
I see, so it sounds like this will require some detective work.

**Comment by ryu0** [comment]
I don't expect that first part to be very hard, but the second part is another story. Do you have any idea what data is stored in these files beyond the obvious text fields?

**Comment by kenorb** [comment]
These files consist of broker-specific data. Each broker got its unique default settings. I'm mainly interested in _Lot step_ or _Lot size_ (float value, such as 0.01) and modifying it, as I believe these values are hardcoded in these files, otherwise, my tests using [FX-MT-VM project fails](https://github.com/EA31337/FX-MT-VM/issues/61) (using [TestLotstep test file](https://github.com/EA31337/FX-MT-VM/blob/master/tests/TestLotstep.mq4)), when I won't load these files.
If you run [MT4 platform](https://github.com/EA31337/FX-MT-VM/releases/download/4.x/mt-4.0.0.1010.zip), it asks you to which server you'd like to connect. So list of servers and company names are coming from this file.

I'm not sure what other data is there, I don't remember exactly what I needed to modify there. The values could be connected to [`server.ini`](https://github.com/FX31337/FX-BT-Scripts/issues/62) which is also unknown. 

I've plenty of other tasks (they can be converted to bounties) which are next pieces of the puzzle. The fields can be currently unknown, can be adjusted later on.

If you get to the binary data, and tell me some info how it looks like, I could have some clues, or it could be not important.

**Comment by ryu0** [comment]
Strange. There's a variable number of bytes between the end of the company name string and the start of the server address string. I'm not sure why this is.

**Comment by ryu0** [comment]
I think I have finished the last of the obvious binary chunks, which all happen to be fixed sized text strings. As for the rest of the binary data, I tried using a hex viewer that could interpret the remaining bytes as various data types, but I was unable to find anything further about their intent or purpose. I could not find any floats or doubles that had a value resembling this "lot step" you mentioned. I think you are right in thinking that the remaining data chunks are obfuscated via an unknown form of data encoding. 

For future reference, I was able to find some hints about the file format from the vServers.h header file from the MT5 C++ API Trial code, located here: http://mtapi.online/download/mt5demo.zip

As this is an undocumented file format, I think I am at the limit of what I can do, but I think I have been able to at least provide you with something that someone else can expand on later. Is this enough for you to consider the original bounty requirements to be satisfied? As you said, I think you will need to split this task into multiple issues to tackle different aspects of the problem, simply because we don't know what other data is lurking in these server files.
